hikari/config.py

```diff
@@ -116,45 +203,182 @@ class HTTPTimeoutSettings:
     """Settings to control HTTP request timeouts."""
 
     acquire_and_connect: typing.Optional[float] = attr.ib(default=None)
-    """Timeout for `request_socket_connect` PLUS connection acquisition."""
+    """Timeout for `request_socket_connect` PLUS connection acquisition.
+
+    By default, this has no timeout allocated.
+
+    Returns
+    -------
+    typing.Optional[builtins.float]
+        The timeout, or `builtins.None` to disable it.
+    """
 
     request_socket_connect: typing.Optional[float] = attr.ib(default=None)
-    """Timeout for connecting a socket."""
+    """Timeout for connecting a socket.
+
+    By default, this has no timeout allocated.
+
+    Returns
+    -------
+    typing.Optional[builtins.float]
+        The timeout, or `builtins.None` to disable it.
+    """
 
     request_socket_read: typing.Optional[float] = attr.ib(default=None)
-    """Timeout for reading a socket."""
+    """Timeout for reading a socket.
+
+    By default, this has no timeout allocated.
+
+    Returns
+    -------
+    typing.Optional[builtins.float]
+        The timeout, or `builtins.None` to disable it.
+    """
 
     total: typing.Optional[float] = attr.ib(default=30.0)
     """Total timeout for entire request.
 
-    Defaults to 30 seconds.
+    By default, this has a 30 second timeout allocated.
+
+    Returns
+    -------
+    typing.Optional[builtins.float]
+        The timeout, or `builtins.None` to disable it.
     """
 
+    @acquire_and_connect.validator
+    @request_socket_connect.validator
+    @request_socket_read.validator
+    @total.validator
+    def _(self, attrib: attr.Attribute[typing.Optional[float]], value: typing.Optional[float]) -> None:
+        # This error won't occur until some time in the future where it will be annoying to
+        # try and determine the root cause, so validate it NOW.
+        if value is not None and (not isinstance(value, (float, int)) or value <= 0):  # type: ignore[unreachable]
```

Comment:
typeshed is nice, it assumes the validators are always getting valid input
that defeats the object of them in the first place surely?